### DISCUSSION TITLE
`isPending` shared between actions

### DISCUSSION THREAD

**Discussion by benjick** [next-safe-action/next-safe-action#295] [Q&A] [answered]
Hey!

Didn't want to open an issue, because I'm not sure if it's an issue.

When I use `isPending` from `useAction` and trigger two different actions (in two different components) it's like their state is "shared".

1. Click action 1
2. Click action 2
3. Both `isPending` equals `true`
4. Action 1 completes
5. Both are still = `true`
6. Action 2 completes
7. Both are = `false`

Is this expected behavior? `isExecuting` does not behave the same way, so I'm guessing it's something with `useTransition`

**Comment by TheEdoRan** [comment, marked as answer]
Hi, yes. That's a technical limitation of React currently. Take a look at this section of the `useTransition` hook docs: https://react.dev/reference/react/useTransition#starttransition-caveats

**Reply by benjick** [reply]
Thanks!
